Add Kong Gateway Operator docs (#6144)
Kong/docs.konghq.com: app/_src/gateway-operator/get-started/konnect/deploy-data-plane.md

              value: pki
            - name: KONG_CLUSTER_CONTROL_PLANE
              value: YOUR_CP_ID.us.cp0.konghq.com:443
            - name: KONG_CLUSTER_SERVER_NAME
              value: YOUR_CP_ID.us.cp0.konghq.com
            - name: KONG_CLUSTER_TELEMETRY_ENDPOINT
              value: YOUR_CP_ID.us.tp0.konghq.com:443
            - name: KONG_CLUSTER_TELEMETRY_SERVER_NAME
              value: YOUR_CP_ID.us.tp0.konghq.com
            - name: KONG_CLUSTER_CERT
              value: /etc/secrets/kong-cluster-cert/tls.crt
            - name: KONG_CLUSTER_CERT_KEY
              value: /etc/secrets/kong-cluster-cert/tls.key
            - name: KONG_LUA_SSL_TRUSTED_CERTIFICATE
              value: system
            - name: KONG_KONNECT_MODE
              value: "on"
            - name: KONG_VITALS
              value: "off"
          volumeMounts:
            - name: cluster-certificate
              mountPath: /var/cluster-certificate
            - name: kong-cluster-cert
              mountPath: /etc/secrets/kong-cluster-cert/
              readOnly: true
        volumes:
          - name: cluster-certificate
          - name: kong-cluster-cert
            secret:

>               secretName: kong-cluster-cert

Reviewer: We're going to need 
```suggestion
              secretName: kong-cluster-cert
              defaultMode: 420
```

until https://github.com/Kong/gateway-operator/issues/904 is fixed

Author: Are you sure? This worked for me when I tested locally

Reviewer: This is tricky, it will "work" but the reconciliation won't continue further than ensuring the Deployment is as expected. So for instance you won't get the `Ready` status condition on the `DataPlane`.

---
title: Deploy a Data Plane
content-type: tutorial
book: kgo-konnect-get-started
chapter: 2
---

In order for {{ site.kgo_product_name }} data planes to attach to {{ site.konnect_short_name }} they need to know which endpoint to connect to, and how to authenticate their requests.

> To get this information, [log in to {{ site.konnect_short_name }}](https://cloud.konghq.com/login) and visit _Gateway Manager_. Select a Control Plane and then create a _New Runtime Instance_.

Reviewer 2: ```suggestion
To get the endpoint and the authentication details of the data plane.
1.  [Log in to {{ site.konnect_short_name }}](https://cloud.konghq.com/login).
1.  Navigate to runtimes icon [Gateway Manager](https://cloud.konghq.com/gateway-manager), choose a control plane, and click **Create a New Data Plane Node**.
1.  In the **Create a Data Plane Node** page select the  **Platform** as *Kubernetes*.
     **Advanced Kubernetes Setup** sections lists the instructions to install Kong Gateway and deploy Kong Gateway data plane nodes in hybrid mode.
1.  Create a namespace named `kong` in the Kubernetes cluster using `kubectl create namespace kong`.   
```

> Follow the instructions on the screen, but stop once you have created a secret in step 3:

Reviewer 2: ```suggestion
1.  Click **Generate Certificates**  in step 3.
1.  Copy and save the contents of **Cluster Certificate** in a file named `tls.crt`.
1.  Copy and save the contents of **Cluster Key** in a file named `tls.key`.

```

> 
> ```bash
> kubectl create secret tls kong-cluster-cert -n kong --cert=/{PATH_TO_FILE}/tls.crt --key=/{PATH_TO_FILE}/tls.key
> ```

Reviewer 2: ```suggestion
1.  To deploy a {{ site.kgo_product_name }} instance, type:
    ```bash
    kubectl create secret tls kong-cluster-cert -n kong --cert=/{PATH_TO_FILE}/tls.crt --key=/{PATH_TO_FILE}/tls.key
    ```
```

> The instructions in {{ site.konnect_short_name }} use Helm to deploy a dataplane instance, but we want to use {{ site.kgo_product_name }}.

Reviewer 2: Delete this sentence

Author: Why should it be deleted? I think it's useful to call out the KGO is a replacement for Helm

> We installed {{ site.kgo_product_name }} in the previous step, which means we can create a `DataPlane` resource directly on the server and the operator will deploy an instance for us.

Reviewer 2: ```suggestion
    You can now create a `DataPlane` resource directly on the server and the operator deploys an instance.
```

> The final step is to find your control plane ID in step 4 of the {{ site.konnect_short_name }} UI. Your control plane ID is the first segment of `cluster_server_name`.
> 
> For example, a `cluster_server_name` of `36fc5d01be.us.cp0.konghq.com` means that my control plane ID is `36fc5d01be`.

Reviewer 2: ```suggestion
    1. In the **Configuration parameters** step 4 of **Advanced Kubernetes Setup**  find the value of `env` defined `cluster_server_name` and the first segment of that value is the control plane ID for your cluster.
    For example, if the value of `cluster_server_name` is `36fc5d01be.us.cp0.konghq.com`, then the control plane ID of the cluster is `36fc5d01be`.
```

> Replace `YOUR_CP_ID` in the manifest below with your control plane ID and apply with `kubectl apply`:

Reviewer 2: ```suggestion
1.  In this maifest, replace `YOUR_CP_ID` with your control plane ID and deploy the data plane with `kubectl apply`:
```

            - name: KONG_CLUSTER_SERVER_NAME
              value: YOUR_CP_ID.us.cp0.konghq.com
            - name: KONG_CLUSTER_TELEMETRY_ENDPOINT
              value: YOUR_CP_ID.us.tp0.konghq.com:443
            - name: KONG_CLUSTER_TELEMETRY_SERVER_NAME
              value: YOUR_CP_ID.us.tp0.konghq.com
            - name: KONG_CLUSTER_CERT
              value: /etc/secrets/kong-cluster-cert/tls.crt
            - name: KONG_CLUSTER_CERT_KEY
              value: /etc/secrets/kong-cluster-cert/tls.key
            - name: KONG_LUA_SSL_TRUSTED_CERTIFICATE
              value: system
            - name: KONG_KONNECT_MODE
              value: "on"
            - name: KONG_VITALS
              value: "off"
          volumeMounts:
            - name: cluster-certificate
              mountPath: /var/cluster-certificate
            - name: kong-cluster-cert
              mountPath: /etc/secrets/kong-cluster-cert/
              readOnly: true
        volumes:
          - name: cluster-certificate
          - name: kong-cluster-cert
            secret:
              secretName: kong-cluster-cert
              defaultMode: 420
' | kubectl apply -f -

> ```

Reviewer 2: This may need to be indented.

> ```

Reviewer 2: ```suggestion
```
    The results should look like this:
    ```text
    dataplane.gateway-operator.konghq.com/dataplane-example configured
    ```

    The data plane that you deployed is added as a node to the control plane. In the **Overview** page of the control plane a message appears that reads: `Congrats, you just created a new Data Plane node! Now, add a gateway service.`
    
```

---
title: Deploy a Data Plane
content-type: tutorial
book: kgo-konnect-get-started
chapter: 2
---

To attach a {{ site.kgo_product_name }} data plane to {{ site.konnect_short_name }} the data plane needs to know which endpoint to connect to, and how to authenticate the requests.

To get the endpoint and the authentication details of the data plane.
1.  [Log in to {{ site.konnect_short_name }}](https://cloud.konghq.com/login).
1.  Navigate to {% konnect_icon runtimes %} [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager), choose the control plane, and click **Create a New Data Plane Node**.

> 1.  In the **Create a Data Plane Node** page select *Kubernetes* as the **Platform**.

Author: Yes, this is how markdown does numbered lists

1.  Create a namespace named `kong` in the Kubernetes cluster
    ```bash
    kubectl create namespace kong
    ```
1.  Click **Generate Certificate**  in step 3.
1. Save the contents of **Cluster Certificate** in a file named `tls.crt`. Save the contents of **Cluster Key** in a file named `tls.key`.
1.  Create a Kubernetes secret containing the cluster certificate:

    ```bash

>     kubectl create secret tls kong-cluster-cert -n kong --cert=/{PATH_TO_FILE}/tls.crt --key=/{PATH_TO_FILE}/tls.key

Author: The `{}` matches what is shown in Konnect

---
title: Deploy a Data Plane
content-type: tutorial
book: kgo-konnect-get-started
chapter: 2
---

To attach a {{ site.kgo_product_name }} data plane to {{ site.konnect_short_name }} the data plane needs to know which endpoint to connect to, and how to authenticate the requests.

To get the endpoint and the authentication details of the data plane.
1.  [Log in to {{ site.konnect_short_name }}](https://cloud.konghq.com/login).
1.  Navigate to {% konnect_icon runtimes %} [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager), choose the control plane, and click **Create a New Data Plane Node**.
1.  In the **Create a Data Plane Node** page select the  **Platform** as *Kubernetes*.

>      The **Advanced Kubernetes Setup** section lists the instructions to install Kong Gateway and deploy Kong Gateway data plane nodes in hybrid mode.

Author: ```suggestion
     The **Advanced Kubernetes Setup** section lists the instructions to install {{ site.base_gateway }} and deploy {{ site.base_gateway }} data plane nodes in hybrid mode.
```

1.  Create a namespace named `kong` in the Kubernetes cluster using `kubectl create namespace kong`.   

1.  Click **Generate Certificates**  in step 3.
1.  Copy and save the contents of **Cluster Certificate** in a file named `tls.crt`.
1.  Copy and save the contents of **Cluster Key** in a file named `tls.key`.

1.  To deploy a {{ site.kgo_product_name }} instance, type:
    ```bash
    kubectl create secret tls kong-cluster-cert -n kong --cert=/{PATH_TO_FILE}/tls.crt --key=/{PATH_TO_FILE}/tls.key
    ```

The instructions in {{ site.konnect_short_name }} use Helm to deploy a dataplane instance, but we want to use {{ site.kgo_product_name }}.

    You can now create a `DataPlane` resource directly on the server and the operator will create an instance of {{ site.base_gateway }}.

    1. In the **Configuration parameters** step 4 of **Advanced Kubernetes Setup** find the value of  `cluster_server_name`. The first segment of that value is the control plane ID for your cluster.
    For example, if the value of `cluster_server_name` is `36fc5d01be.us.cp0.konghq.com`, then the control plane ID of the cluster is `36fc5d01be`.

> 1.  In this maifest, replace `YOUR_CP_ID` with your control plane ID and deploy the data plane with `kubectl apply`:

Author: ```suggestion
1.  In this manifest, replace `YOUR_CP_ID` with your control plane ID and deploy the data plane with `kubectl apply`:
```
